projects/dev/src/app/shared/models/group.model.ts

```diff
@@ -1,20 +1,59 @@
-export interface Group {
+import { ManagementLevel } from '../constants/group';
+
+export class Group {
   id: string;
   name: string;
   type: string;
   grade: number;
   description: string;
-  activity_id: string;
-  code: string;
-  code_lifetime: string;
-  code_expires_at: Date;
+  code?: string;
+  code_lifetime?: string;
+  code_expires_at?: Date;
 
   current_user_is_manager: boolean;
   current_user_is_member: boolean;
+  current_user_can_grant_group_access: boolean;
+  current_user_can_manage: string;
+  current_user_can_watch_members: boolean;
```

Comment:
I don't know how you determined a part of your attributes are optional but that does not match what is returned by the backend. Some attributes may not be given by the backend, indicated by the missing "required" in the output format in the [spec](https://france-ioi.github.io/algorea-devdoc/api/#operation/groupGet). 
Maybe you did it on purpose but I doubt it.

Reply:
fixed.